### PR TITLE
Refactor: Replace lambdas with comprehensions

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

File: airflow/jobs/backfill_job_runner.py [hunk 1]
```diff
@@ -742,7 +742,7 @@ def tabulate_ti_keys_set(ti_keys: Iterable[TaskInstanceKey]) -> str:
 
             if all(key.map_index == -1 for key in ti_keys):
                 headers = ["DAG ID", "Task ID", "Run ID", "Try number"]
-                sorted_ti_keys = (k[0:4] for k in sorted_ti_keys)
+                sorted_ti_keys = [k[0:4] for k in sorted_ti_keys]
```
Comment (uranusjr) [inline airflow/jobs/backfill_job_runner.py]: This does not seem to fit in this PR?

Comment (eumiro) [reply airflow/jobs/backfill_job_runner.py]: Oh yes, thanks, removed.

File: dev/breeze/src/airflow_breeze/utils/parallel.py [hunk 2]
```diff
@@ -357,7 +357,7 @@ def print_async_summary(completed_list: list[ApplyResult]) -> None:
 
 def get_completed_result_list(results: list[ApplyResult]) -> list[ApplyResult]:
     """Return completed results from the list."""
-    return list(filter(lambda result: result.ready(), results))
+    return [result for result in results if result.ready]
```
Comment (uranusjr) [inline dev/breeze/src/airflow_breeze/utils/parallel.py]:
```suggestion
    return [result for result in results if result.ready()]
```

File: dev/example_dags/update_example_dags_paths.py [hunk 3]
```diff
@@ -102,19 +102,18 @@ def find_matches(_file: Path, provider: str, version: str):
...
+    dirs: list[Path] = [p for p in curdir.iterdir() if p.is_dir()]
     with Progress(console=console) as progress:
         task = progress.add_task(f"Updating {len(dirs)}", total=len(dirs))
         for directory in dirs:
             if directory.name.startswith("apache-airflow-providers-"):
                 provider = directory.name[len("apache-airflow-providers-") :]
                 console.print(f"[bright_blue] Processing {directory}")
-                version_dirs = list(filter(os.path.isdir, directory.iterdir()))
+                version_dirs = [p for p in directory.iterdir() if p.is_dir()]
                 for version_dir in version_dirs:
```
Comment (uranusjr) [inline dev/example_dags/update_example_dags_paths.py]:
```python
for version_dir in directory.iterdir():
    if not version_dir.is_dir():
        continue
    ...
```

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
You need to make it ready for review @eumiro

**Pull request merged**